azure-kusto-ingest/source/streamingIngestClient.js

```diff
@@ -0,0 +1,68 @@
+const KustoClient = require("azure-kusto-data").Client;
+const { FileDescriptor, StreamDescriptor } = require("./descriptors");
+const DataFormat = require("./resourceManager"); 
+const zlib = require('zlib');
+const fs = require('fs');
+
+module.exports = class KustoStreamingIngestClient {
+    constructor(kcsb, defaultProps) {
+        this.kustoClient = new KustoClient(kcsb);
+        this.defaultProps = defaultProps;
+        this._mapping_required_formats = [ DataFormat.JSON, DataFormat.SINGLEJSON, DataFormat.AVRO ];
```

Comment:
```suggestion
        this._mapping_required_formats = Object.freeze([ DataFormat.JSON, DataFormat.SINGLEJSON, DataFormat.AVRO ]);
```

This make it readonly. (another option is using `new Set()`